script.js

```diff
@@ -252,17 +252,80 @@ const saveEditFromInput = async (index, inputShopEl, inputDateEl, inputMoneyEl,
       },
       body: JSON.stringify({
         id: allExpenses[index]._id,
-        company: inputShopEl.value.trim(),
-        date: inputDateEl.value,
-        money: inputMoneyEl.value
+        company: inputShopValue.trim(),
+        date: inputDateValue,
+        money: inputMoneyValue
       })
     });
 
-    allExpenses[index].company = inputShopEl.value.trim();
-    allExpenses[index].money = Number(inputMoneyEl.value);
-    allExpenses[index].date = inputDateEl.value;
+    allExpenses[index].company = inputShopValue.trim();
+    allExpenses[index].money = Number(inputMoneyValue);
+    allExpenses[index].date = inputDateValue;
 
-    if (value) allExpenses[index].isEdit = !allExpenses[index].isEdit;
+    allExpenses[index].isEdit = !allExpenses[index].isEdit;
+  } else {
+    deleteElement(index);
+  };
+  render();
+};
+
+const saveCompanyFromDBInput = async (index, companyValue) => {
+  if (companyValue.trim()) {
+    const resp = await fetch(`http://localhost:8000/editExpense`, {
```

Comment:
can you make an edit request into a separate function that accepts an object for sending as a parameter? due to this, you will shorten your code